verify-contracts.sh

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env bash
+set -euo pipefail
```

Comment:
TBH I dislike when we have files that are hard to review manually, most of our codebase/scripts are in TypeScript. Considering that logs are generated inside l1-contracts repo and l1-contracts repo already have ts files, I would prefer it to be a ts script, than a bash one